internal/pkg/checkin/bulk.go

```diff
@@ -202,6 +202,11 @@ func (bc *Bulk) flush(ctx context.Context) error {
 			if pendingData.extra.ver != "" {
 				fields[dl.FieldAgent] = map[string]interface{}{
 					dl.FieldAgentVersion: pendingData.extra.ver,
+					// Duplicates the writes done when an upgrade is ack'd by the agent.
+					// This helps eliminate the case where the agent is stuck in upgrading after it's already upgraded if the ack
+					// is never received.
+					dl.FieldUpgradeStartedAt: nil,
+					dl.FieldUpgradedAt: nowTimestamp,
 				}
```

Comment:
So every agent will be upgradedAt at now() by default on any ack?

Reply:
besides it'll lose the `UpgradeStartedAt` as well.

Reply:
No because this logic only executes when the version has changed